In [1]:
import pandas as pd
ad_features = pd.read_csv('ad_feature.csv')
ad_features
del ad_features

adgroup_id  cate_id  campaign_id  customer     brand   price
0            63133     6406        83237         1   95471.0  170.00
1           313401     6406        83237         1   87331.0  199.00
2           248909      392        83237         1   32233.0   38.00
3           208458      392        83237         1  174374.0  139.00
4           110847     7211       135256         2  145952.0   32.99
...            ...      ...          ...       ...       ...     ...
846806      824255     4526       380022    255868  389713.0  268.00
846807      790170     4280       379736    255872  322171.0   68.00
846808      845286     6261       379736    255872  322171.0   88.00
846809      824732     4520       379736    255872  322171.0   68.00
846810      845337    11156       379603    255874   74120.0  279.00

[846811 rows x 6 columns]

In [2]:
import pandas as pd
raw_sample = pd.read_csv('raw_sample.csv')
raw_sample
del raw_sample

user  time_stamp  adgroup_id          pid  nonclk  clk
0         581738  1494137644           1  430548_1007       1    0
1         449818  1494638778           3  430548_1007       1    0
2         914836  1494650879           4  430548_1007       1    0
3         914836  1494651029           5  430548_1007       1    0
4         399907  1494302958           8  430548_1007       1    0
...          ...         ...         ...          ...     ...  ...
26557956    6834  1494139949      846808  430548_1007       1    0
26557957   16815  1494328475      846808  430548_1007       1    0
26557958  736609  1494597871      846809  430548_1007       1    0
26557959  735781  1494684550      846809  430539_1007       1    0
26557960  735781  1494677729      846810  430539_1007       1    0

[26557961 rows x 6 columns]

In [3]:
import pandas as pd
user_profile = pd.read_csv('user_profile.csv')
user_profile
del user_profile

userid  cms_segid  cms_group_id  final_gender_code  age_level  \
0            234          0             5                  2          5   
1            523          5             2                  2          2   
2            612          0             8                  1          2   
3           1670          0             4                  2          4   
4           2545          0            10                  1          4   
...          ...        ...           ...                ...        ...   
1061763  1137955          0             3                  2          3   
1061764  1138545          0             4                  2          4   
1061765  1138902         67             9                  1          3   
1061766  1139632          0             7                  1          1   
1061767  1139870          8             2                  2          2   

         pvalue_level  shopping_level  occupation  new_user_class_level   
0                 NaN               3           0                    3.0  
1                 1.0               3           1                    2.0  
2                 2.0               3           0                    NaN  
3                 NaN               1           0                    NaN  
4                 NaN               3           0                    NaN  
...               ...             ...         ...                    ...  
1061763           NaN               3           0                    4.0  
1061764           NaN               3           0                    3.0  
1061765           2.0               3           0                    4.0  
1061766           NaN               1           0                    NaN  
1061767           2.0               3           0                    3.0  

[1061768 rows x 9 columns]

In [4]:
import pandas as pd
UserBehavior = pd.read_csv('UserBehavior.csv')
UserBehavior
del UserBehavior

1  2268318  2520377  pv  1511544070
0               1  2333346  2520771  pv  1511561733
1               1  2576651   149192  pv  1511572885
2               1  3830808  4181361  pv  1511593493
3               1  4365585  2520377  pv  1511596146
4               1  4606018  2735466  pv  1511616481
...           ...      ...      ...  ..         ...
100150801  999999  4797808    11120  pv  1512293403
100150802  999999  4613472  4602841  pv  1512293766
100150803  999999  3647364  2304296  pv  1512293792
100150804  999999  1903801  2304296  pv  1512293827
100150805  999999  3696094  4602841  pv  1512293891

[100150806 rows x 5 columns]

In [9]:
import pandas as pd

# Load the datasets
ecommerce_users = pd.read_csv("UserBehavior.csv", usecols=["1"])
ad_users = pd.read_csv("user_profile.csv", usecols=["userid"])

# Convert to sets for fast intersection
ecommerce_users_set = set(ecommerce_users["1"])
ad_users_set = set(ad_users["userid"])

# Find common users
common_users = ecommerce_users_set.intersection(ad_users_set)

# Count total users in e-commerce dataset
total_ecommerce_users = len(ecommerce_users_set)

# Compute percentage
percent_common_users = (len(common_users) / total_ecommerce_users) * 100

# Print results
print(f"Number of common users: {len(common_users)}")
print(f"Percentage of common users: {percent_common_users:.2f}%")


Number of common users: 916856
Percentage of common users: 92.80%


In [18]:
import pandas as pd

# Load the datasets
ecommerce_users = pd.read_csv("UserBehavior.csv", usecols=["1"])
ad_users = pd.read_csv("user_profile.csv", usecols=["userid"])

# Convert to sets to get unique users
ecommerce_users_set = set(ecommerce_users["1"])  # Unique users in e-commerce dataset
ad_users_set = set(ad_users["userid"])  # Unique users in advertising dataset

# Find common users
common_users = ecommerce_users_set.intersection(ad_users_set)

# Compute correct percentage (dividing by the number of unique users, not interactions)
percent_common_users = (len(common_users) / len(ecommerce_users_set)) * 100




Number of common users: 916856
Total unique users in e-commerce dataset: 987994
Percentage of common users: 92.80%


In [ ]:
ad_users = pd.read_csv("user_profile.csv", usecols=["userid"])

In [21]:
ad_users = pd.read_csv("raw_sample.csv", usecols=["user"])

# Convert to sets to get unique users
ecommerce_users_set = set(ecommerce_users["1"])  # Unique users in e-commerce dataset
ad_users_set = set(ad_users["user"])  # Unique users in advertising dataset

# Find common users
common_users = ecommerce_users_set.intersection(ad_users_set)

# Compute correct percentage (dividing by the number of unique users, not interactions)
percent_common_users = (len(common_users) / len(ecommerce_users_set)) * 100
# Print results
print(f"Number of common users: {len(common_users)}")
print(f"Total unique users in e-commerce dataset: {len(ecommerce_users_set)}")
print(f"Total unique users in ads Users dataset: {len(ad_users_set)}")
print(f"Percentage of common users: {percent_common_users:.2f}%")

Number of common users: 987994
Total unique users in e-commerce dataset: 987994
Total unique users in ads Users dataset: 1141729
Percentage of common users: 100.00%


In [20]:
# Print results
print(f"Number of common users: {len(common_users)}")
print(f"Total unique users in e-commerce dataset: {len(ecommerce_users_set)}")
print(f"Total unique users in ads Users dataset: {len(ad_users_set)}")
print(f"Percentage of common users: {percent_common_users:.2f}%")

Number of common users: 916856
Total unique users in e-commerce dataset: 987994
Total unique users in ads Users dataset: 1061768
Percentage of common users: 92.80%


In [17]:
del ecommerce_users
del ad_users
del ecommerce_users_set
del ad_users_set
del common_users
del percent_common_users

In [13]:
percent_common_users = len(common_users) / len(ecommerce_users) * 100
print(f"Pourcentage d'utilisateurs communs : {percent_common_users:.2f}%")

Pourcentage d'utilisateurs communs : 0.92%


In [12]:
print("the users common : ", len(common_users))

the users common :  916856


In [7]:
import pandas as pd

# Charger les datasets
ecommerce_category = pd.read_csv("UserBehavior.csv", usecols=["2520377"])
ad_category = pd.read_csv("ad_feature.csv", usecols=["brand"])

# Vérifier l'intersection des utilisateurs
common_category = set(ecommerce_category["2520377"]).intersection(set(ad_category["brand"]))

# Calculer le pourcentage d'utilisateurs communs
percent_common_category = len(common_category) / len(ecommerce_category) * 100
print(f"Pourcentage des category communs : {percent_common_category:.2f}%")


Pourcentage des category communs : 0.00%


In [6]:
UserBehavior.columns = ['user_id', 'item_id', 'category_id', 'behavior_type', 'timestamp']
UserBehavior

user_id  item_id  category_id behavior_type   timestamp
0                1  2333346      2520771            pv  1511561733
1                1  2576651       149192            pv  1511572885
2                1  3830808      4181361            pv  1511593493
3                1  4365585      2520377            pv  1511596146
4                1  4606018      2735466            pv  1511616481
...            ...      ...          ...           ...         ...
100150801   999999  4797808        11120            pv  1512293403
100150802   999999  4613472      4602841            pv  1512293766
100150803   999999  3647364      2304296            pv  1512293792
100150804   999999  1903801      2304296            pv  1512293827
100150805   999999  3696094      4602841            pv  1512293891

[100150806 rows x 5 columns]

In [7]:
# Charger les datasets
ecommerce_items = pd.read_csv("UserBehavior.csv")
ecommerce_items.columns = ['user_id', 'item_id', 'category_id', 'behavior_type', 'timestamp']



In [8]:
ad_items = pd.read_csv("ad_feature.csv", usecols=["brand"])

# Vérifier l'intersection des items
common_items = set(ecommerce_items["item_id"]).intersection(set(ad_items["adgroup_id"]))

# Calculer le pourcentage d'items communs
percent_common_items = len(common_items) / len(ecommerce_items) * 100
print(f"Pourcentage d'items communs : {percent_common_items:.2f}%")

Pourcentage d'items communs : 0.68%


In [3]:
import pandas as pd

# Charger les datasets
ecommerce_data = pd.read_csv("UserBehavior.csv", usecols=["1", "2268318"])
ad_data = pd.read_csv("raw_sample.csv", usecols=["user", "adgroup_id"])

# Vérifier les utilisateurs communs
ecommerce_users = set(ecommerce_data["1"])
ad_users = set(ad_data["user"])
common_users = ecommerce_users.intersection(ad_users)

# Filtrer les interactions des utilisateurs communs
ecommerce_filtered = ecommerce_data[ecommerce_data["1"].isin(common_users)]
ad_filtered = ad_data[ad_data["user"].isin(common_users)]

# Vérifier les items communs
ecommerce_items = set(ecommerce_filtered["2268318"])
ad_items = set(ad_filtered["adgroup_id"])
common_user_item_interactions = ecommerce_items.intersection(ad_items)

# Calcul du pourcentage
percent_common_user_item = len(common_user_item_interactions) / len(ecommerce_filtered) * 100
print(f"Pourcentage d'interactions utilisateur-item communes : {percent_common_user_item:.2f}%")


Pourcentage d'interactions utilisateur-item communes : 0.65%


In [8]:
del ecommerce_data
del ad_data
del ecommerce_users
del ad_users
del common_users
del ecommerce_filtered
del ad_filtered
del common_user_item_interactions
del percent_common_user_item

NameError: name 'ecommerce_data' is not defined

In [29]:
# Vérifier si les utilisateurs ET les items sont communs
common_users_items = ecommerce_users[ecommerce_users["user_id"].isin(common_users)]
common_users_items = common_users_items[common_users_items["item_id"].isin(common_items)]

# Calculer le pourcentage
percent_common_users_items = len(common_users_items) / len(ecommerce_users) * 100
print(f"Pourcentage d'interactions utilisateur-item communes : {percent_common_users_items:.2f}%")


Pourcentage d'interactions utilisateur-item communes : 16.41%


In [20]:
del ecommerce_users

In [24]:
from scipy.stats import pearsonr

# Charger les taux d'achat et taux de clics
ecommerce_behavior = pd.read_csv("UserBehavior.csv", usecols=["pv"])



In [26]:
del ecommerce_behavior

In [25]:
import pandas as pd

# Charger les datasets
ecommerce_data = pd.read_csv("UserBehavior.csv", usecols=["user_id", "item_id"])
ad_data = pd.read_csv("raw_sample.csv", usecols=["user", "adgroup_id"])

# Vérifier les utilisateurs communs
ecommerce_users = set(ecommerce_data["user_id"])
ad_users = set(ad_data["user"])
common_users = ecommerce_users.intersection(ad_users)

# Filtrer les interactions des utilisateurs communs
ecommerce_filtered = ecommerce_data[ecommerce_data["user_id"].isin(common_users)]
ad_filtered = ad_data[ad_data["user"].isin(common_users)]

# Vérifier les items communs
ecommerce_items = set(ecommerce_filtered["item_id"])
ad_items = set(ad_filtered["adgroup_id"])
common_user_item_interactions = ecommerce_items.intersection(ad_items)

# Calcul du pourcentage
percent_common_user_item = len(common_user_item_interactions) / len(ecommerce_filtered) * 100
print(f"Pourcentage d'interactions utilisateur-item communes : {percent_common_user_item:.2f}%")


ValueError: x and y must have length at least 2.

# Calcule the Similarity

In [5]:
import pandas as pd

# Load user IDs from both datasets
ecommerce_users = pd.read_csv("UserBehavior.csv")


In [4]:
ecommerce_users.columns = ['user_id', 'item_id', 'category_id', 'behavior_type', 'timestamp']
ad_users = pd.read_csv("raw_sample.csv")

# Convert to sets for unique user counts
ecommerce_users_set = set(ecommerce_users["user_id"])
ad_users_set = set(ad_users["user"])

# Compute common users
common_users = ecommerce_users_set.intersection(ad_users_set)

# Compute percentage
percent_common_users = (len(common_users) / len(ecommerce_users_set)) * 100

# Print results
print(f"Number of common users: {len(common_users)}")
print(f"Total unique users in e-commerce dataset: {len(ecommerce_users_set)}")
print(f"Percentage of common users: {percent_common_users:.2f}%")


Number of common users: 987994
Total unique users in e-commerce dataset: 987994
Percentage of common users: 100.00%


In [10]:
print(f"Total unique users in ads dataset: {len(ad_users_set)}")
print(f"Total unique users in e-commerce dataset: {len(ecommerce_users_set)}")


Total unique users in ads dataset: 1141729
Total unique users in e-commerce dataset: 987994


# Similarity Items

In [6]:
# Convert to sets for unique item counts
ecommerce_items_set = set(ecommerce_users["item_id"])
ad_items_set = set(ad_users["adgroup_id"])

# Compute common items
common_items = ecommerce_items_set.intersection(ad_items_set)

# Compute percentage
percent_common_items = (len(common_items) / len(ecommerce_items_set)) * 100

# Print results
print(f"Number of common items: {len(common_items)}")
print(f"Total unique items in e-commerce dataset: {len(ecommerce_items_set)}")
print(f"Percentage of common items: {percent_common_items:.2f}%")

Number of common items: 682020
Total unique items in e-commerce dataset: 4162024
Percentage of common items: 16.39%


In [7]:
print(f"total unique items in ads group : {len(ad_items_set)}")

total unique items in ads group : 846811


# Notes : 
- on a presque 4 millions items en dataset target
- on a presque 846811 items en dataset source

# User-Item Interaction Similarity (Common User-Item Pairs)

In [8]:
# Convert to sets for unique user-item pairs
ecommerce_pairs = set(zip(ecommerce_users["user_id"], ecommerce_users["item_id"]))
ad_pairs = set(zip(ad_users["user"], ad_users["adgroup_id"]))

# Compute common user-item interactions
common_user_item_interactions = ecommerce_pairs.intersection(ad_pairs)

# Compute percentage
percent_common_user_item = (len(common_user_item_interactions) / len(ecommerce_pairs)) * 100

# Print results
print(f"Number of common user-item interactions: {len(common_user_item_interactions)}")
print(f"Total unique user-item interactions in e-commerce dataset: {len(ecommerce_pairs)}")
print(f"Percentage of common user-item interactions: {percent_common_user_item:.2f}%")


Number of common user-item interactions: 318
Total unique user-item interactions in e-commerce dataset: 75969638
Percentage of common user-item interactions: 0.00%


In [9]:
print(f"Total unique user-item interactions in e-commerce dataset: {len(ecommerce_pairs)}")
print(f"Total unique user-item interactions in ad_paires dataset: {len(ad_pairs)}")
print(f"Total unique user-item interactions in e-commerce dataset: {len(common_user_item_interactions)}")

Total unique user-item interactions in e-commerce dataset: 75969638
Total unique user-item interactions in ad_paires dataset: 23195671
Total unique user-item interactions in e-commerce dataset: 318


In [13]:
ecommerce_categories_set = set(ecommerce_users["category_id"])



In [14]:
ad_users = pd.read_csv('ad_feature.csv')
ad_categories_set = set(ad_users["cate_id"])
common_categories = ecommerce_categories_set.intersection(ad_categories_set)
percent_common_categories = (len(common_categories) / len(ecommerce_categories_set)) * 100
print(f"Percentage of common categories: {percent_common_categories:.2f}%")

Percentage of common categories: 0.14%


In [15]:
print(f"Total unique categories interactions in e-commerce dataset: {len(ecommerce_categories_set)}")
print(f"Total unique categories in ads dataset: {len(ad_categories_set)}")
print(f"Common categories: {len(common_categories)}")
print(f"Percentage of common categories: {percent_common_categories:.2f}%")

Total unique categories interactions in e-commerce dataset: 9439
Total unique categories interactions in ads dataset: 6769
Common categories: 13
Percentage of common categories: 0.14%
